src/logion/model/locrequest.model.ts

```diff
@@ -248,6 +266,36 @@ export class LocRequestAggregateRoot {
         }));
     }
 
+    preVoid(reason: string) {
+        if(this.voidInfo !== undefined && this.voidInfo.reason !== null && this.voidInfo.reason !== undefined) {
+            throw new Error("LOC is already void")
+        }
+        this.voidInfo = new EmbeddableVoidInfo();
+        this.voidInfo.reason = reason;
+    }
+
+    voidLoc(timestamp: Moment) {
+        if(this.voidInfo !== undefined && this.voidInfo.voidedOn !== undefined && this.voidInfo.voidedOn !== null) {
+            throw new Error("LOC is already void");
+        }
+        if(this.voidInfo === undefined) {
+            this.voidInfo = new EmbeddableVoidInfo();
+            this.voidInfo.reason = "";
+        }
+        this.voidInfo.voidedOn = timestamp.toISOString();
+    }
+
+    getVoidInfo(): VoidInfo | null {
+        if(this.voidInfo !== undefined && this.voidInfo.reason !== null) {
+            return {
+                reason: this.voidInfo.reason || "",
+                voidedOn: (this.voidInfo.voidedOn !== undefined && this.voidInfo.voidedOn !== null) ? moment(this.voidInfo.voidedOn) : null
```

Comment:
I believe this could be simplified to:
```suggestion
                voidedOn: this.voidInfo.voidedOn ? moment(this.voidInfo.voidedOn) : null
```

Reply:
Indeed.